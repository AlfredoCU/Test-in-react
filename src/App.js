import React from "react";
import PropTypes from "prop-types";

const App = ({ dev, age, date }) => {
  // console.log(dev, age, date);
  return (
    <>
      <h1>Hola Mundo</h1>
      <p>React - {dev}</p>
      <p>Edad: {age}</p>
      <b>Cumpleaños: {date}</b>
    </>
  );
};

//* Las propTypes, nos ayudan a solicitar un tipo de dato en especifico que recibira el componente.
//* También podemos forzar a que las props sean requeridas o obligatorias.
App.propTypes = {
  dev: PropTypes.string.isRequired,
  age: PropTypes.number,
  date: PropTypes.string, //* Aunque el valor sea por defecto, se le puede asignar un tipo de dato.
};

//* Valores por defecto.
App.defaultProps = {
  date: "11/02//1997",
};

export default App;
