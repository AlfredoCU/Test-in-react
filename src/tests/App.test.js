import React from "react";
// import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import App from "../App";

describe("Prueba a <App />", () => {
  // test("Debe de renderizar el componente", () => {
  //   const name = "Alfredo CU";
  //   // const wrapper = render(<App dev="Alfredo CU" age={23} />);
  //   // wrapper.getByText()
  //   const { getByText } = render(<App dev={name} age={23} />);
  //   expect(getByText("React - " + name)).toBeInTheDocument();
  // });

  test("Prueba a <App />", () => {
    const name = "Alfredo CU";
    const wrapper = shallow(<App dev={name} age={23} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar la fecha mostrada por props", () => {
    const name = "Alfredo CU";
    const date = "11/02//1997";
    const wrapper = shallow(<App dev={name} age={23} date={date} />);
    //! Funciona sólo para un nodo, si existe más etiquetas b entonces mostara un error.
    const texto = wrapper.find("b").text();
    expect(texto).toBe(`Cumpleaños: ${date}`);
  });
});
