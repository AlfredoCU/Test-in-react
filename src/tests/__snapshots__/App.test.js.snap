// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a <App /> Prueba a <App /> 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    React - 
    Alfredo CU
  </p>
  <p>
    Edad: 
    23
  </p>
  <b>
    Cumplea√±os: 
    11/02//1997
  </b>
</Fragment>
`;
